### PR TITLE
refactor: change button to get updates and use loading animation [JB-211] [JB-212]

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

Closes https://linear.app/peel/issue/JB-212/add-loading-animation-for-subscribe-button

Closes https://linear.app/peel/issue/JB-211/add-subscribe-text-to-subscribe-button

## Screenshots or screen recordings

[Screen Recording 2023-04-04 at 2.57.55 pm.mov](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/pinTSPPZHK80UCzWAYgo/9e4b3bdf-611c-4bf5-ba22-4b2c3b290c61/Screen%20Recording%202023-04-04%20at%202.57.55%20pm.mov)

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] (if relevant) I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] (if relevant) I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3347** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3347" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3347?utm_source=stack-comment).

File: src/components/SubscribeButton/SubscribeButton.tsx [hunk 1]
```diff
@@ -40,11 +32,18 @@ const _SubscribeButton = ({
...
+            className,
+          )}
           type="text"
           onClick={onSubscribeButtonClicked}
+          loading={loading}
+          icon={<SubscribeButtonIcon isSubscribed={isSubscribed} />}
         >
-          <SubscribeButtonIcon isSubscribed={isSubscribed} />
+          <span className="font-base text-sm">
+            <Trans>Get Updates</Trans>
```
Comment (tomquirk) [inline src/components/SubscribeButton/SubscribeButton.tsx:45]: WDYT about `Watch` instead? Feel like this is common enough language these days.

File: src/components/SubscribeButton/SubscribeButton.tsx [hunk 2]
```diff
@@ -40,11 +32,18 @@ const _SubscribeButton = ({
         }
       >
         <Button
-          className={twMerge('p-0', className)}
+          className={twMerge(
+            'flex items-center justify-center gap-3 p-0',
```
Comment (tomquirk) [inline src/components/SubscribeButton/SubscribeButton.tsx:36]: I think the `p-0` here might be a bit squishy when the button is in active/loading state:

**Review by tomquirk — CHANGES_REQUESTED** [review]
See inline comments

File: src/components/SubscribeButton/SubscribeButton.tsx (hunk 2 above)
Comment (wraeth-eth) [reply src/components/SubscribeButton/SubscribeButton.tsx:36]: agree, but this is antd at work unfortunately. We'd have to rebuild the button I think to get it behaving right

Comment (tomquirk) [reply src/components/SubscribeButton/SubscribeButton.tsx:36]: can't we just nuke p-0?

Comment (wraeth-eth) [reply src/components/SubscribeButton/SubscribeButton.tsx]: done

**Pull request merged**